### PR TITLE
feat: teach PyArray to compare

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

File: pyvortex/src/array.rs [hunk 1]
```diff
@@ -138,6 +138,71 @@ impl PyArray {
...
+    /// >>> a.compare(b, 'lt').to_arrow_array()
+    /// <pyarrow.lib.BooleanArray object at ...>
+    /// [
+    ///   true,
+    ///   null,
+    ///   false,
+    ///   false,
+    ///   false
+    /// ]
+    fn compare(&self, other: &Bound<PyArray>, operator: &Bound<PyString>) -> PyResult<PyArray> {
```
Comment (robert3005) [inline pyvortex/src/array.rs]: I think this is a very weird python api, would this be better as a individual funtions instead of string switch?

Comment (danking) [reply pyvortex/src/array.rs]: I agree it is weird. PyArrow doesn't implement __lt__ and friends but, AFAICT, nobody has complained on their issues about it. They do implement `__eq__`. In the compute functions API, they do provide [`greater`](https://arrow.apache.org/docs/python/generated/pyarrow.compute.greater.html) and friends.

My best guess is that those operations are allocating so they make them available through an API that exposes the memory pool parameter. It seems like that's [a way to use](https://arrow.apache.org/docs/cpp/memory.html#default-memory-pool) jemalloc or mimalloc for Arrow arrays even if the rest of the application uses some other allocator. It seems like that could be used to create an Arena for arrow arrays and do O(1) free after, say, processing a chunk of data.

Anyway, I think you're right that we should implement the operators. Even if we later provide a way to specify an allocator, the operators can continue to use the default allocator.

Comment (robert3005) [reply pyvortex/src/array.rs]: I was mostly even thinking that even if it's not the `__eq__` or `__compare__` I think it's better to get rid of stringly typed operator. FWIW the C api takes an int for operator

**Review by robert3005 — APPROVED** [review]

File: pyvortex/src/array.rs (hunk 1 above)
Comment (danking) [reply pyvortex/src/array.rs]: I switched to operators and moved the docs into the examples for technical reasons (see comment).

**Pull request merged**